src/recording/macromanager.h

```diff
@@ -0,0 +1,80 @@
+#pragma once
+
+#include <QObject>
+
+#include "engine/enginemaster.h"
+#include "mixer/playermanager.h"
+#include "preferences/usersettings.h"
+#include "recording/recordingmanagerbase.h"
+
+#define MAX_MACRO_SIZE 1000
+#define NO_DECK nullptr
+
+struct MacroAction {
```

Comment:
Not that important for now, but please use classes for stuff that isn't in the anonymous name space.

Reply:
Could you add that to the coding guidelines in the wiki? I wanted to know your policy for structs but didn't find any.

Isn't this the exact use case struts where meant for?

Reply:
In C, yes. In C++, structs are just regular classes with default member visibility public instead of private.

Reply:
Yeah, but default visibility public makes total sense for these things, or not? They are almost plain data structures.

Also, please document in https://github.com/mixxxdj/mixxx/wiki/coding_guidelines